### ISSUE TITLE
Error: The Ticker must be disposed before flutter: calling super.dispose

### ISSUE THREAD

**Issue by felixwortmann** [fluttercommunity/backdrop#83]
**Describe the bug**
I get the following error:
```
flutter: The following assertion was thrown while finalizing the widget tree:
flutter: BackdropScaffoldState#ba6f2(ticker active) was disposed with an active Ticker.
flutter: BackdropScaffoldState created a Ticker via its SingleTickerProviderStateMixin, but at the time
flutter: dispose() was called on the mixin, that Ticker was still active. The Ticker must be disposed before
flutter: calling super.dispose().
flutter: Tickers used by AnimationControllers should be disposed by calling dispose() on the
flutter: AnimationController itself. Otherwise, the ticker will leak.
flutter: The offending ticker was:
flutter:   Ticker(created by BackdropScaffoldState#ba6f2(lifecycle state: created))
flutter:   The stack trace when the Ticker was actually created was:
flutter:   #0      new Ticker.<anonymous closure> (package:flutter/src/scheduler/ticker.dart:67:40)
flutter:   #1      new Ticker (package:flutter/src/scheduler/ticker.dart:69:6)
flutter:   #2      SingleTickerProviderStateMixin.createTicker
flutter:   (package:flutter/src/widgets/ticker_provider.dart:129:15)
flutter:   #3      new AnimationController (package:flutter/src/animation/animation_controller.dart:247:21)
flutter:   #4      BackdropScaffoldState.initState (package:backdrop/scaffold.dart:397:9)
```
When I follow the stacktrace, I get to this point in `BackdropScaffoldState`
```dart
@override
  void dispose() {
    super.dispose();
    if (_shouldDisposeAnimationController) _animationController.dispose();
  }
```

If I change that to :
```dart
@override
  void dispose() {
    _animationController.dispose();
    super.dispose();
  }
```
everything works again.<br>

I now ask myself why  `_shouldDisposeAnimationController` is false if an AnimationController exists? 
Also I found this in `BackdropScaffoldState`'s `initState`<br>
```dart
if (widget.animationController == null && widget.controller == null) {
      _shouldDisposeAnimationController = true;
    }
```
Since `_shouldDisposeAnimationController` otherwise defaults to false, `_animationController.dispose()` is not called every time.<br>
I think it could just be changed to:
```dart
@override
  void dispose() {
    if(_animationController != null) _animationController.dispose();
    super.dispose();
  }
```

<br><br>
**Smartphone**
 - Device: iPhone 12
 - OS: IOS 14.4

**Comment by WieFel** [comment]
@felixwortmann you are right.

Actually I ask myself why `_shouldDisposeAnimationController` is needed at all. In my opinion it could also be removed, and the dispose method changed as follows:
```dart
@override
  void dispose() {
    _animationController?.dispose();
    super.dispose();
  }
```
It should work as well, wouldn't it? @daadu

**Comment by felixwortmann** [comment]
@WieFel that is exactly what I meant. I already created a pr, you can just try it out.

**Comment by daadu** [comment]
If you pass `animationController` to `BackdropScaffold`, the `dispose` responsibility is of the outer widget. Ideally this is how it should be.

The Ticker was created according to logs in `BackdropScaffoldState.initState (package:backdrop/scaffold.dart:397:9)` which is the following:

https://github.com/fluttercommunity/backdrop/blob/3c6da7e32243ff7c4d59b35e73c426f78e9e5628/lib/scaffold.dart#L394-L404

So, the `_animationController` is initialized inside the widget state (and not passed) in that case, `_shouldDisposeAnimationController` will be `true` and will be disposed of, when the widget disposes.

**Comment by daadu** [comment]
@felixwortmann "some how" the dispose is not being called, my suspect is in the `BackdropScaffoldState.dispose`, `super.dispose` is called before `controller.dispose`.

Simply, interchanging that should work. The `dispose` of `BackdropScaffoldState` Should be the following :
```dart
  @override
  void dispose() {
    if (_shouldDisposeAnimationController) _animationController.dispose();
   super.dispose(); // this one at last
  }
```

**Comment by daadu** [comment]
@felixwortmann Can you confirm, if the above code works for you?

**Comment by felixwortmann** [comment]
> @felixwortmann Can you confirm, if the above code works for you?

That works for me. `super.dispose()` should just be called after 
```dart
if (_shouldDisposeAnimationController) _animationController.dispose();
```

**Comment by felixwortmann** [comment]
I just updated my PR, can you take a look at it?

**Comment by daadu** [comment]
Fixed with #84

**Comment by daadu** [comment]
This fix/feature has landed with [v0.5.3 on pub.dev](https://pub.dev/packages/backdrop/versions/0.5.3).
